src/components/ScenarioParameters/components/ScenarioParametersTabs.js

```diff
@@ -30,23 +31,109 @@ const useStyles = makeStyles((theme) => ({
       fontWeight: 'bold',
       color: theme.palette.primary.contrastText,
     },
+    '& .MuiTab-wrapper': {
+      '& .MuiSvgIcon-root': {
+        marginLeft: '5px',
+      },
+      flexDirection: 'row-reverse',
+    },
   },
   placeholder: {
     margin: `0 ${theme.spacing(3)}px`,
   },
 }));
 
-const ScenarioParametersTabs = ({ parametersGroupsMetadata }) => {
+function _buildScenarioTabList(tabs, userRoles, classes, t) {
+  const tabListComponent = [];
+  for (const groupMetadata of tabs) {
+    const lockedTab = !hasRequiredProfile(userRoles, groupMetadata.authorizedRoles);
+    if (lockedTab) {
+      if (groupMetadata.showParameterGroupIfNoPermission) {
+        tabListComponent.push(
+          <Tab
+            key={groupMetadata.id}
+            value={groupMetadata.id}
+            data-cy={groupMetadata.id + '_tab'}
+            icon={<LockIcon />}
+            className={classes.tab}
+            label={t(TranslationUtils.getParametersGroupTranslationKey(groupMetadata.id), groupMetadata.id)}
+          />
+        );
+      }
+    } else {
+      tabListComponent.push(
+        <Tab
+          key={groupMetadata.id}
+          value={groupMetadata.id}
+          data-cy={groupMetadata.id + '_tab'}
+          className={classes.tab}
+          label={t(TranslationUtils.getParametersGroupTranslationKey(groupMetadata.id), groupMetadata.id)}
+        />
+      );
+    }
+  }
+  return tabListComponent;
+}
+
+function _buildTabPanels(userRoles, tabs, classes) {
+  const tabPanelComponents = [];
+  for (let index = 0; index < tabs.length; index++) {
+    const groupMetadata = tabs[index];
+    const lockedTab = !hasRequiredProfile(userRoles, groupMetadata.authorizedRoles);
+    if (lockedTab) {
+      if (groupMetadata.showParameterGroupIfNoPermission) {
+        tabPanelComponents.push(
+          <TabPanel index={index} key={groupMetadata.id} value={groupMetadata.id} className={classes.tabPanel}>
+            {groupMetadata.tab}
+          </TabPanel>
+        );
+      }
+    } else {
+      tabPanelComponents.push(
+        <TabPanel index={index} key={groupMetadata.id} value={groupMetadata.id} className={classes.tabPanel}>
+          {groupMetadata.tab}
+        </TabPanel>
+      );
```

Comment:
To prevent duplicating code, could simplify these conditions into :
```
if (!lockedTab || !groupMetadata.showParameterGroupIfNoPermission) {
  tabPanelComponents.push(  [...]  );
}
```